x/fundraising/types/auction.go

```diff
@@ -372,12 +372,12 @@ func VestingReserveAcc(auctionId uint64) sdk.AccAddress {
 	return DeriveAddress(ReserveAddressType, ModuleName, VestingReserveAccPrefix+AccNameSplitter+fmt.Sprint(auctionId))
 }
 
-// IsAuctionStarted returns true if the start time of the auction is equal or ahead of the given time t.
+// IsAuctionStarted returns true if the start time of the auction is equal or before the given time t.
 func IsAuctionStarted(startTime time.Time, t time.Time) bool {
 	return !startTime.After(t)
 }
 
-// IsAuctionFinished returns true if the end time of the auction is equal or ahead of the given time t.
+// IsAuctionFinished returns true if the end time of the auction is equal or before the given time t.
 func IsAuctionFinished(endTime time.Time, t time.Time) bool {
-	return !endTime.Before(t)
+	return !endTime.After(t)
```

Comment:
I think _Before_ is right. Is there a reason you changed it?

Reply:
As I write test code for the invariant check, I found it odd in [this code](https://github.com/tendermint/fundraising/pull/57/files#diff-ab5c7ce62048705a75675bc524d0b28100859011ce83e4a46e9b36634f7bbd3cL87). Although I added -1 day, the auction updates to be finished. Therefore, I fixed it.

If you look at the updated [test cases](https://github.com/tendermint/fundraising/pull/57/files#diff-ec166810572d1867337711b8400b97533b01510fb9cee0069a006d24395c41c4R36-R38), I think it makes sense that `IsAuctionFinished` returns trues when the end time is equal or before the current time.
 
For example, end time is `2021-11-01T00:00:00Z` and current time is `2021-12-10T00:00:00Z`. `IsAuctionFinished` returns true since the end time is already pass the current time.

Reply:
I understood, I think it can be confusing as the test code increases by `StartTime, EndTime`. For a more intuitive understanding, it would be better to add a case where `BlockTime(now)` increases. for `IsStarted` and `IsFinished`

Also, if we refactor the functions like below, We can recycle the function for Auction types with multiple `EndTimes`.

```go
func (ba BaseAuction) IsStarted(t time.Time) bool {
	return !ba.GetStartTime().After(t)
}

func (ba BaseAuction) IsFinished(t time.Time) bool {
	endTimes := ba.GetEndTimes()
	return !endTimes[len(endTimes)-1].After(t)
}
```